src/overlays/actors/ovl_En_Baguo/z_en_baguo.c

```diff
@@ -27,92 +38,458 @@ const ActorInit En_Baguo_InitVars = {
+    // the Nejiron should forcibly explode and as a loop index.
+    i = false;
+    if (this->action != NEJIRON_ACTION_EXPLODING && this->action != NEJIRON_ACTION_RETREATING) {
+        if (!(this->actor.bgCheckFlags & 1) && this->actor.world.pos.y < (this->actor.home.pos.y - 100.0f)) {
+            // Force a detonation if we're off the ground and have fallen
+            // below our home position (e.g., we rolled off a ledge).
+            i = true;
+        }
+        if (this->actor.bgCheckFlags & 0x60 && this->actor.depthInWater >= 40.0f) {
+            // Force a detonation if we're too far below the water's surface.
+            i = true;
+        }
+        if ((this->collider.base.acFlags & AC_HIT || i)) {
+            this->collider.base.acFlags &= ~AC_HIT;
+            if (i || this->actor.colChkInfo.damageEffect == NEJIRON_DMGEFF_KILL) {
+                func_800BCB70(&this->actor, 0x4000, 0xFF, 0, 8);
+                this->action = NEJIRON_ACTION_EXPLODING;
+                this->actor.speedXZ = 0.0f;
+                this->actor.shape.shadowScale = 0.0f;
+
+                for (i = 0; i < ARRAY_COUNT(this->particles); i++) {
+                    acceleration.x = (Rand_ZeroOne() - 0.5f) * 8.0f;
+                    acceleration.y = -1.0f;
+                    acceleration.z = (Rand_ZeroOne() - 0.5f) * 8.0f;
+                    velocity.x = (Rand_ZeroOne() - 0.5f) * 14.0f;
+                    velocity.y = Rand_ZeroOne() * 30.0f;
+                    velocity.z = (Rand_ZeroOne() - 0.5f) * 14.0f;
+                    EnBaguo_InitializeParticle(this, &this->actor.focus.pos, &velocity, &acceleration,
+                                               (Rand_ZeroFloat(1.0f) * 0.01f) + 0.003f, 90);
+                }
+
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, this->actor.world.pos.x,
+                            this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, CLEAR_TAG_POP);
+                Audio_PlayActorSound2(&this->actor, NA_SE_IT_BOMB_EXPLOSION);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_BAKUO_DEAD);
+                this->timer = 30;
+                this->actor.flags |= 0x8000000;
+                this->actor.flags &= ~1;
+                Actor_SetScale(&this->actor, 0.0f);
+                this->collider.elements->dim.scale = 3.0f;
+                this->collider.elements->info.toucher.damage = 8;
+                Item_DropCollectibleRandom(globalCtx, NULL, &this->actor.world.pos, 0xB0);
+                this->actionFunc = EnBaguo_PostDetonation;
+            }
+        }
+    }
+}
+
+void EnBaguo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBaguo* this = THIS;
+
+    Actor_SetHeight(&this->actor, 30.0f);
+    EnBaguo_UpdateParticles(this, globalCtx);
+    EnBaguo_CheckForDetonation(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+
+    if (this->blinkTimer != 0) {
+        this->blinkTimer--;
```

Comment:
Does DECR macro match these two timer decrements?

Reply:
It does indeed match